userdata.sh.tmpl

```diff
@@ -19,6 +19,7 @@ sleep 5
 
 # Start tailscale
 sudo tailscale up \
-    --advertise-routes=${routes} \
-    --authkey=${authkey} \
-    --hostname=${hostname}
+  --advertise-routes=${routes} \
+  --advertise-tags=${tags} \
```

Comment:
I'm confused about this change. I assume something could be changed in Tailscale auth/tagging logic, but previously the machine was tagged when authenticated with the pre-authentication Tailnet key, see the [comment](https://github.com/masterpointio/terraform-aws-tailscale/blob/d7a7f56d06b4beb1466bd58a6fe52c59c384d225/main.tf#L49).
I get why we might want to concat context.tf tags + additional tags provided by a user, but I don't get why we need to add this flag and use different tags for the key and for `up` command.

Reply:
To recap the conversation we had over a call for history purposes: 

1. We weren't including the `primary_tag` in this list, which was a mistake on my end. I've now fixed that in a recent push. 
2. We're going to keep this `--advertise-tags` bit here because it's useful to be explicit since Tailscale is confusing af at times.